src/commands/publish/mod.rs

```diff
@@ -89,7 +89,23 @@ fn publish_script(user: &GlobalUser, target: &Target) -> Result<(), failure::Err
 fn upload_buckets(target: &Target, user: &GlobalUser) -> Result<(), failure::Error> {
     for namespace in &target.kv_namespaces() {
         if let Some(bucket) = &namespace.bucket {
+            if bucket.is_empty() {
+                failure::bail!(
+                    "{} You need to specify a bucket directory in your wrangler.toml",
+                    emoji::WARN
+                )
+            }
             let path = Path::new(&bucket);
+            if !path.exists() {
+                if let Some(path_str) = path.to_str() {
+                    failure::bail!(
+                        "{} could not find a directory for bucket \"{}\"",
+                        emoji::WARN,
```

Comment:
what if it's a file but not a directory?

Reply:
oh good point

Reply:
bucket must be a directory.